### PR TITLE
Add testpmd Unit tests

### PR DESCRIPTION
**Pull request by RamLavi**
This PR is adding unit tests to check the testpmd stats parsing.

### CODE REVIEW

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
+package testpmd_test
```
Comment (orelmisan) [inline pkg/internal/checkup/executor/testpmd/console_test.go:1]: Could you please add the license comment?

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+const (
+	vmiUnderTestEastNICPCIAddress = "0000:06:00.0"
+	trafficGenEastMACAddress      = "60:94:19:c9:ac:01"
+	vmiUnderTestWestNICPCIAddress = "0000:07:00.0"
+	trafficGenWestMACAddress      = "60:94:19:c9:ac:02"
+	verbosePrintsEnabled          = false
+)
+
+func TestGetPortStatsSuccess(t *testing.T) {
+	t.Run("when GetStats succeed", func(t *testing.T) {
```
Comment (orelmisan) [inline pkg/internal/checkup/executor/testpmd/console_test.go:24]: Please consider removing the sub-test if there is only one.

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+		expecter := stubExpecter{}
+		c := testpmd.NewTestpmdConsole(expecter, vmiUnderTestEastNICPCIAddress, trafficGenEastMACAddress,
+			vmiUnderTestWestNICPCIAddress, trafficGenWestMACAddress, verbosePrintsEnabled)
```
Comment (orelmisan) [inline pkg/internal/checkup/executor/testpmd/console_test.go:27]: nit:
```suggestion
		c := testpmd.NewTestpmdConsole(
		expecter,
		vmiUnderTestEastNICPCIAddress,
		trafficGenEastMACAddress,
		vmiUnderTestWestNICPCIAddress,
		trafficGenWestMACAddress,
		verbosePrintsEnabled,
		 )
```

Same in the following occurrences.

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+				TXTotal:   480000018,
+			},
+		}
+		assert.Equal(t, expected, stats, "GetStats returned unexpected result")
+	})
+}
+
+func TestGetPortStatsFailure(t *testing.T) {
+	t.Run("when fails on batch error", func(t *testing.T) {
+		expectBatchErr := errors.New("failed to run batch")
```
Comment (orelmisan) [inline pkg/internal/checkup/executor/testpmd/console_test.go:63]:
```suggestion
		expectedBatchErr := errors.New("failed to run batch")
```

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+				TXDropped: 17,
+				TXTotal:   480000018,
+			},
+		}
+		assert.Equal(t, expected, stats, "GetStats returned unexpected result")
+	})
+}
+
+func TestGetPortStatsFailure(t *testing.T) {
+	t.Run("when fails on batch error", func(t *testing.T) {
```
Comment (orelmisan) [inline pkg/internal/checkup/executor/testpmd/console_test.go:62]:
```suggestion
	t.Run("when batch execution fails", func(t *testing.T) {
```

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+		}
+
+		c := testpmd.NewTestpmdConsole(expecter, vmiUnderTestEastNICPCIAddress, trafficGenEastMACAddress,
+			vmiUnderTestWestNICPCIAddress, trafficGenWestMACAddress, verbosePrintsEnabled)
+		stats, err := c.GetStats()
+
+		assert.ErrorContains(t, err, expectBatchErr.Error())
+		assert.Empty(t, stats)
+	})
+	t.Run("when fails on timeout", func(t *testing.T) {
```
Comment (orelmisan) [inline pkg/internal/checkup/executor/testpmd/console_test.go:75]:
```suggestion
	t.Run("when batch times out", func(t *testing.T) {
```

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+		expectTimeoutErr := errors.New("failed on timeout")
```
Comment (orelmisan) [inline pkg/internal/checkup/executor/testpmd/console_test.go:76]:
```suggestion
		expectedTimeoutErr := errors.New("failed on timeout")
```

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+		expecter := &stubExpecter{
+			timeoutErr: expectTimeoutErr,
+		}
+		c := testpmd.NewTestpmdConsole(expecter, vmiUnderTestEastNICPCIAddress, trafficGenEastMACAddress,
+			vmiUnderTestWestNICPCIAddress, trafficGenWestMACAddress, verbosePrintsEnabled)
+		stats, err := c.GetStats()
+
+		assert.ErrorContains(t, err, expectTimeoutErr.Error())
```
Comment (orelmisan) [inline pkg/internal/checkup/executor/testpmd/console_test.go:84]: nit:
```suggestion

		stats, err := c.GetStats()
		assert.ErrorContains(t, err, expectTimeoutErr.Error())
```

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi 
WDYT about testing `Run()` and `ClearStats()`?

File: pkg/internal/checkup/executor/testpmd/console_test.go (hunk 1 above)
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:1]: DONE

File: pkg/internal/checkup/executor/testpmd/console_test.go (hunk 2 above)
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:24]: DONE

File: pkg/internal/checkup/executor/testpmd/console_test.go (hunk 3 above)
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:27]: DONE

File: pkg/internal/checkup/executor/testpmd/console_test.go (hunk 5 above)
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:62]: DONE

File: pkg/internal/checkup/executor/testpmd/console_test.go (hunk 4 above)
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:63]: DONE

File: pkg/internal/checkup/executor/testpmd/console_test.go (hunk 6 above)
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:75]: DONE

File: pkg/internal/checkup/executor/testpmd/console_test.go (hunk 7 above)
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:76]: DONE

File: pkg/internal/checkup/executor/testpmd/console_test.go (hunk 8 above)
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:84]: DONE

File: pkg/internal/checkup/executor/testpmd/console_test.go [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+		assert.Empty(t, stats)
+	})
+}
+
+type stubExpecter struct {
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply pkg/internal/checkup/executor/testpmd/console_test.go:89]: DONE

**Comment by RamLavi** [comment]
> Thank you for the PR @RamLavi WDYT about testing `Run()` and `ClearStats()`?

I don't think there is anything to unit test there..

**Review by orelmisan — APPROVED** [review]
Thank you

**Pull request merged**